pinot-common/src/test/java/com/linkedin/pinot/common/data/FieldSpecTest.java

```diff
@@ -192,6 +195,69 @@ public void testTimeFieldSpecConstructor() {
   }
 
   /**
+   * Test {@link TimeFieldSpec} constructors.
+   */
+  @Test
+  public void testDateTimeFieldSpecConstructor() {
+    String name = "Date";
+    DataType dataType = DataType.LONG;
+    String format = "1:HOURS:EPOCH";
+    String granularity = "1:HOURS";
+    int defaultNullValue = 17050;
+
+    DateTimeFieldSpec dateTimeFieldSpec1 = new DateTimeFieldSpec(name, dataType, format, granularity);
+    DateTimeFieldSpec dateTimeFieldSpec2 = new DateTimeFieldSpec(name, dataType, format, granularity, defaultNullValue);
+
+    Assert.assertFalse(dateTimeFieldSpec1.equals(dateTimeFieldSpec2));
+    dateTimeFieldSpec1.setDefaultNullValue(defaultNullValue);
+    Assert.assertEquals(dateTimeFieldSpec1, dateTimeFieldSpec2);
+
+    DateTimeFieldSpec dateTimeFieldSpec3 = new DateTimeFieldSpec(name, dataType, format, granularity, DateTimeType.PRIMARY);
+    DateTimeFieldSpec dateTimeFieldSpec4 = new DateTimeFieldSpec(name, dataType, format, granularity, DateTimeType.PRIMARY, defaultNullValue);
+
+    Assert.assertFalse(dateTimeFieldSpec3.equals(dateTimeFieldSpec4));
+    dateTimeFieldSpec3.setDefaultNullValue(defaultNullValue);
+    Assert.assertEquals(dateTimeFieldSpec3, dateTimeFieldSpec4);
+
+  }
+
+  @Test(expectedExceptions = {IllegalArgumentException.class})
```

Comment:
If you get IllegalArgumentException in line 233, the other lines will not be executed, right? Isnt it better to actually test for the exception in the line in which you expect it to happen?

Reply:
Fixed